tests/integration/test_auth.py

```diff
@@ -79,21 +79,34 @@ def test_auth_can_create_authenticated_requests_sessions():
     assertions.assertTrue("Bearer" in session.headers["Authorization"])
 
 
-def test_auth_can_fetch_s3_credentials():
+@pytest.mark.parametrize(
+    ("daac", "provider"),
+    (
+        ("NSIDC", "NSIDC_CPRD"),
+        ("GHRCDAAC", "GHRC_DAAC"),
+        ("PODAAC", "POCLOUD"),
+        ("ASF", "ASF"),
+        ("ORNLDAAC", "ORNL_CLOUD"),
+        ("LPDAAC", "LPCLOUD"),
+        ("GES_DISC", "GES_DISC"),
+        ("OBDAAC", "OB_CLOUD"),
+        ("LAADS", "LAADS"),
+        ("ASDC", "LARC_CLOUD"),
+    ),
+)
+def test_auth_can_fetch_s3_credentials(daac, provider):
     activate_environment()
     auth = earthaccess.login(strategy="environment")
     assertions.assertTrue(auth.authenticated)
-    for daac in earthaccess.daac.DAACS:
-        if len(daac["s3-credentials"]) > 0:
-            try:
-                logger.info(f"Testing S3 credentials for {daac['short-name']}")
```

Comment:
Since the value of `daac` will be a string, subscript access will throw an error.

Currently integration tests are silently failing, even though they give a green check. It looks like this has been happening for a while. I'm investigating. #737 

Once that's fixed, then we can continue to debug this test :)

Reply:
Ok so I think we should wait till then

Reply:
#738 resolves the issue, hopefully we can merge the change soon, but even after that there are errors blocking this test from executing. If you are interested, I think some tests need to be refactored to use fixtures.

Reply:
Yeah but which one's need refactoring. If you could provide some info then maybe I can proceed :)

Reply:
There are a couple refactors we need:

* Tests which import `assertions` and call for example `assertions.assertTrue(x)` should be updated to builtin assertions, for example `assert x is True`.
* Tests which do setup at the module level should be updated to use [fixtures](https://docs.pytest.org/en/6.2.x/fixture.html)

I think these should probably be done separately if you want to take them on; the first one would probably be the best place to start IMO because it will give us better error messages.

This test module demonstrates both things I think we're looking to change: https://github.com/nsidc/earthaccess/blob/main/tests/integration/test_onprem_open.py

Perhaps the best place to start though is to see if we already have issues open for this, and if not, create one. If you can help out with that, it would be awesome :)

Reply:
I might create an issue but my descriptions are going to pretty vague( I am not good at explaining at all).